### DISCUSSION TITLE
RTL (Right to left) support

### DISCUSSION THREAD

**Discussion by HarelM** [community/community#2894] [Discussions]
My community speaks Hebrew.
They are intimidated by the need to write in English in Github.
Besides the fact that I can't customize the discussion UI to be in Hebrew the support for RTL is not good.
One needs to know arbitrary stuff such as writing < div dir="rtl" > in the beginning of the post.
I'd love to move the discussions to be in Github and provide Q&A and documentation of solutions etc.
This currently can't be done when the users are afraid of the Github site and it's extremely hard to read Hebrew text when not formatted correctly.
Our current platform is Facebook which is horrible in terms of search, defining a post that represents the solution etc, but the Hebrew support there is much better than here...
I hope this will be solved sometime, I've opened an issue about better RTL support in Github issues a long time ago...
For my project, discussions are not relevant without proper RTL support.

**Comment by becca** [comment]
@HarelM 👋 as someone who grew up speaking and reading hebrew I understand but I imagine our experiences are quite different so any additional information you could provide here would be helpful. 

Does your community currently host its code on GitHub but conversations are what's on Facebook? May I ask what specific aspects are the most painful for you right now?

**Reply by ahangarha** [reply]
It is very clear that if I want to write RTL text, it should appear RTL not LTR. If Gitlab supports this, why github doesn't? Because one one hand they don't care and on the other hand they are not FLOSS themselves so the FLOSS community itself can solve this stupid issue.

The best approach is to use a playform which is truly FLOSSy not as marketing strategy.

**Comment by HarelM** [comment]
I'm hosting my code here:
https://github.com/IsraelHikingMap/Site
Site/App users (who might not be github user for that matter) find it hard to use github for discussions and issue reporting. They prefer to report issues or usage questions in our Facebook here:
https://www.facebook.com/groups/israelhikingmap/
When I politely ask them to open the issue in github they prefer not to do it and I have a hard time tracking issues in Facebook (it's a horrible platform for that).
We have created a FAQ page to address this, but I think the discussion in Github can significantly improve the experience around common issues and pitfalls (users can mark something as a solution and we can have a living database of FAQ and solutions form them, allowing the community to help solve most of the issues they know of already:
https://github.com/IsraelHikingMap/Site/wiki/Israel-Hiking-Maps-FAQ

This was a long intro... I would split this feature request into two unrelated features:
1. Support text alignment automatically for each paragraph/line. This can be seen when writing a mail in gmail. When starting to write a sentence the editor automatically knows to which end to align the text. This should also be applied to viewing, not just editing.
For example I expect the following line to be aligned right both when editing and when viewing:
זאת שורה שכתובה בעברית
2. Allow customization or language change of the discussion area in github - currently it only supports English and only left to right UI. 
It would be great if I could select a language for this area and have it in Hebrew and from Right to left.
Both will reduce the fear of users to get in here and write their comments, ideas, problems, solutions etc.
I don't expect to replace Facebook, but a more language specific interface for the discussions area would, I believe, allow easier relations with the community.
In my perspective and wishful thinking, I would like to remove the FAQ page and have everything in the discussions - this will allow searching which is currently not available for mobile users easily, and better experience when you have an issue or a question. It will allow users to help and remove the need for us to maintain a FAQ page, I hope...

I truly love github. The missing RTL support it not something I believe should be very complex to implement and support and I think it would do great things to communities around Github.

Thanks for a great product!

**Comment by HarelM** [comment]
Here's a recent example from our Facebook group:
https://www.facebook.com/groups/israelhikingmap/permalink/3538275432894291/
![image](https://user-images.githubusercontent.com/3269297/102005485-c5ff7380-3d21-11eb-8757-46730012663b.png)
To translate the above:
"Sorry, I can't get along with Github, I can't find where to write feature proposals..."
While not exactly the issue at hand, I believe a language specific UI for discussions (and/or for Github as a whole) might help a bit...

**Comment by HarelM** [comment]
Also please check out the following issue:
https://github.com/dear-github/dear-github/issues/147
There are a lot of people in that thread with valuable info - all of them would like this to simply be solved...
It would mean a lot to us if some of the features related to RTL world be implemented...

**Comment by hashemi-soroush** [comment]
Can we get an update on this, @becca ? A timeline would be great, but a simple status report will do, too, since it's not marked as accepted or rejected or anything.

**Reply by becca** [reply]
@Sayed-Soroush-Hashemi this is not a priority at the moment for the Discussions product. I'll sync up with other product owners to make sure there's awareness of this request in case it helps with (re)prioritization. 🙇

**Reply by ahangarha** [reply]
> this is not a priority at the moment

Hearing this from Github whose identity is built on top of being FLOSS supporter is so funny. Let me explain it more. this means:

*Oh! FLOSS community come to our proprietary platform on which you have no control and cannot solve your basic problems on, and let me earn by hosting you FLOSS projects on my proprietary one and laugh at your. We decide what has priority and wont even let you to serve us as free labor to fix the problem of thousands of people like you.*

We should host our code where they respect FLOSS culture. Continuing to host code on github means forcing others to remain on it. Cut this loop.

**Reply by hashemi-soroush** [reply]
@becca 
thanks for the quick update. Can we know what percentage of Github repos actually need this feature? This can give us a hint as to where this feature is on the priority list and when we can expect it.

By the way, does Github publish reports like this? There are certain things that I think the community likes to know. For example, what percentage of Github users are using the newly introduced dark mode.

**Reply by hashemi-soroush** [reply]
@ahangarha 
Your attitude doesn't help. Gitlab supports this feature, so you can move your project(s) there, but if you want to stay, it's your decision. If you want this feature so bad, try proving this feature is a priority, for example by presenting some statistics about how many Github repos need this or a comparison of the RTL-repo to LTR-repo ratio in Github and Gitlab.

**Reply by ahangarha** [reply]
@Sayed-Soroush-Hashemi You have full right to try your best but for me, begging companies to implement feature on their proprietary software (as a service) is not an option at all. I wont even waste my time to prove the importance of any feature. I prefer to dedicate that time to help some FLOSS project. I have not been an dedicated programmer till now but that much I was, I moved my hosting somewhere else. I don't use Github as code hosting yet because of other projects, I didn't close my account so that I can make issue and send PR.

You can still try but keep it in your mind that till the time Github remain proprietary software and tries to earn from FLOSS but not comply with FLOSS culture, the main issue remains alive: you have no other option than beggin them to do what you could do in just few minutes. If you are willing to take such insult, go on. I wont do that and I try to motivate more people and projects to leave such platform.

And notice the fact that the request for adding RTL/bidi support has turned into 5 years old. You might be successful in fixing this issue by increasing its priority for Github dev team but the issue will pass from this to another. Till Github is proprietary, there will be no true solution.

**Comment by SaadBazaz** [comment]
Yo @Github, what's the holdup? This is such a small fix but it's taking 5 years?
Do you not realize that 75% of the world has no understanding of basic English? How do you expect to make software "free for the world" when you blatantly ignore such a huge population?

**Comment by ahangarha** [comment]
It seems they have implemented bidi support. You may check it yourself. if works perfectly fine on any md file as well as texts in issues.

یک two سه four

though the text editor doesn't support it yet!

I hope they come with and article explaining why and how they decided to finally add this feature.
